qualtran/_infra/registers.py

```diff
@@ -49,15 +52,16 @@ class Register:
 
     Attributes:
         name: The string name of the register
-        bitsize: The number of (qu)bits in the register.
+        dtype: Quantum data type this register is representing. For example, an
+            array of 32-bit fixed-point floats.
         shape: A tuple of integer dimensions to declare a multidimensional register. The
             total number of bits is the product of entries in this tuple times `bitsize`.
         side: Whether this is a left, right, or thru register. See the documentation for `Side`
             for more information.
     """
 
     name: str
-    bitsize: int
+    dtype: QDTypeT
```

Comment:
we should write an attrs validator that raises a helpful error message if an integer is passed here.